packages/office-addin-manifest/src/export.ts

```diff
@@ -31,7 +38,36 @@ async function createZip(manifestPath: string): Promise<AdmZip> {
     throw new Error(`The file '${manifestPath}' does not exist`);
   }
 
-  const manifest: devPreview.DevPreviewSchema = await ManifestUtil.loadFromPath(manifestPath);
+  const manifest: DevPreviewSchema = readJsonFileSync(manifestPath);
+  const agents = manifest?.copilotAgents?.declarativeAgents;
+
+  if (agents) {
+    agents.forEach((agent, agentIndex) => {
+      const file: string = agent?.file;
+      const filePath = path.join(manifestDir, file);
+
+      if (!fs.existsSync(filePath)) {
+        throw new Error(
+          `copilotAgents.declarativeAgents[${agentIndex}].file does not exist:  ${filePath}`
+        );
+      }
+
+      zip.addLocalFile(filePath, "", file);
+      const agentJson: DeclarativeCopilotManifestSchema = readJsonFileSync(filePath);
+
+      agentJson?.actions?.forEach((action, actionIndex) => {
+        if (action?.file) {
+          const actionFilePath = path.join(manifestDir, action.file);
+          if (!fs.existsSync(actionFilePath)) {
+            throw new Error(`actions[${actionIndex}].file does not exist: ${actionFilePath}`);
+          }
+
+          zip.addLocalFile(actionFilePath, "", action.file);
```

Comment:
We need to maintain the path structure of the dependent files inside the zip, otherwise cases where there are other folders in the path for the agent files will break because the file will have been relocated in the zip folder.  See the icon files as an example.

Same is true of the base agent path.